src/unitxt/operators.py

```diff
@@ -508,75 +508,121 @@ def process_value(self, value: Any) -> Any:
+        The added prefix or suffix will be of length 1 - 5 patterns. The actual length is also randomly selected.
+     remove_existing_whitespaces : allows to first clean any existing leading and trailing whitespaces.
+        The selected pattern(s) are then prepended and/or appended to the potentially trimmed input.
 
 
     Examples:
-        to append a '\n' or a '\t' to the end of the input, employ
-        AugmentSuffix(augment_model_input=True, suffixes=['\n','\t'])
-        If '\n' is preferred over '\t', at 2:1 ratio, employ
-        AugmentSuffix(augment_model_input=True, suffixes={'\n':2,'\t':1})
-        which will append '\n' twice as often as '\t'.
+        To prepend the input with a prefix made of 1 to 5 '\n'-s or '\t'-s, employ
+        AugmentPrefixSuffix(augment_model_input=True, prefixes=['\n','\t'], suffixes = None)
+        To append the input with a suffix made of 1 to 5 '\n'-s or '\t'-s, with an all '\n'-s suffixes
+        being preferred over all '\t'-s, at 2:1 ratio, employ
+        AugmentPrefixSuffix(augment_model_input=True, suffixes={'\n':2,'\t':1}, prefixes = None)
+        which will append '\n'-s twice as often as '\t'-s.
+        The actual length of the prefix and/or suffix is randomly selected, uniformly over [1..5]
+        If only one of prefixes/suffixes is needed, set the other to None.
 
     """
 
-    suffixes: Optional[Union[List[str], Dict[str, int]]] = [" ", "\n", "\t"]
-    remove_existing_trailing_whitespaces: Optional[bool] = False
+    prefixes: Optional[Union[List[str], Dict[str, int]]] = {
+        " ": 20,
+        "\\t": 10,
+        "\\n": 40,
+        "": 30,
+    }
+    suffixes: Optional[Union[List[str], Dict[str, int]]] = {
+        " ": 20,
+        "\\t": 10,
+        "\\n": 40,
+        "": 30,
+    }
+    remove_existing_whitespaces: Optional[bool] = False
 
     def verify(self):
         assert (
-            isinstance(self.suffixes, list) or isinstance(self.suffixes, dict)
-        ), f"Argument 'suffixes' should be either a list or a dictionary, whereas it is of type {type(self.suffixes)}"
-
-        if isinstance(self.suffixes, dict):
-            for k, v in self.suffixes.items():
-                assert isinstance(
-                    k, str
-                ), f"suffixes should map strings, whereas key {k!s} is of type {type(k)}"
-                assert isinstance(
-                    v, int
-                ), f"suffixes should map to ints, whereas value {v!s} is of type {type(v)}"
-        else:
-            for k in self.suffixes:
-                assert isinstance(
-                    k, str
-                ), f"suffixes should be a list of strings, whereas member {k!s} is of type {type(k)}"
+            self.prefixes or self.suffixes
+        ), "At least one of prefixes/suffixes should be not None."
+        for arg, arg_name in zip(
+            [self.prefixes, self.suffixes], ["prefixes", "suffixes"]
```

Comment:
Elron has a better way for type checking.  @elronbandel - can you please advise?

Reply:
added relevant comment in the main thread